src/main/java/com/atlassian/db/replica/api/reason/Reason.java

```diff
@@ -8,26 +8,28 @@
 public final class Reason {
     private final String name;
     private final boolean isRunOnMain;
+    private final boolean isWrite;
 
-    private Reason(final String name, boolean isRunOnMain) {
+    private Reason(final String name, boolean isRunOnMain, boolean isWrite) {
         this.name = name;
         this.isRunOnMain = isRunOnMain;
+        this.isWrite = isWrite;
     }
 
-    public static final Reason RW_API_CALL = new Reason("RW_API_CALL", true);
-    public static final Reason REPLICA_INCONSISTENT = new Reason("REPLICA_INCONSISTENT", true);
-    public static final Reason READ_OPERATION = new Reason("READ_OPERATION", false);
-    public static final Reason WRITE_OPERATION = new Reason("WRITE_OPERATION", true);
-    public static final Reason LOCK = new Reason("LOCK", true);
+    public static final Reason RW_API_CALL = new Reason("RW_API_CALL", true, true);
+    public static final Reason REPLICA_INCONSISTENT = new Reason("REPLICA_INCONSISTENT", true, false);
+    public static final Reason READ_OPERATION = new Reason("READ_OPERATION", false, false);
+    public static final Reason WRITE_OPERATION = new Reason("WRITE_OPERATION", true, true);
+    public static final Reason LOCK = new Reason("LOCK", true, false);
     public static final Reason MAIN_CONNECTION_REUSE = new Reason(
         "MAIN_CONNECTION_REUSE",
-        true
-    );
+        true,
```

Comment:
At the end turns out yes!!!  Reason can point out SQL write queries. Though it might happen that not write will still be run on Main due to connection state/ transaction. The reason is derivative from couple of variables (statement,SQL, connection) and it used for couple of reason to build :

-  read/write statement, 
-  read/write connection
And at the same time RouteDecision is used to determine statement/connection(state) 

In future new combinations of SQL /statement/connection should end up building new Reasons.
e.g. now we have LOCK => only for `SELECT FOR UPDATE` ( we may have more like this)

It's much more explicit right know on compile lvl